bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RetryingReadRowsOperation.java

```diff
@@ -292,6 +294,16 @@ private void handleTimeoutError(Status status) {
     }
   }
 
+  protected CallOptions getCallOptions() {
+    if (CallOptionsFactory.ConfiguredCallOptionsFactory.isGet(nextRequest)) {
+      // This is a get, so apply a finite deadline.
+      return super.getCallOptions();
```

Comment:
super.getCallOptions() vs super.getUserCallOptions() is confusing, perhaps explain the difference?